[VPR] Add options to disable range checks in AoE combo

memorial36b opened the pull request:
Removes the action range checks from the Vicepit combo sequence and Reawaken in the AoE combo (simple and advanced), so they will still fire without a target selected.

Akechi-kun commented:
Why, if I may ask? 

This could cause issues for users who mindlessly hold the one-button without checking which skills are being used. 

_Vicepit_ & it's combo route _should_ only be used when in range of a target, even if you have to hold one coil from one mob to another as it's better to do this than to just waste the skill & damage altogether because you weren't within range. 

This is also notably bad for _Reawaken_, since you _want_ to be in range of the target to get the 750 potency from the hit regardless of any circumstance.

I do not recommend this PR.

memorial36b commented:
I have added options to disable the range checks. The only changes I have made to the logic itself is that I have changed the melee range checks to 5y range checks, as I believe that is fair given the Vicepit combo's 5y radius.

I would argue that at minimum the Vicepit combo _excluding Vicepit itself_ should have its range checks removed, as currently if you lose your target while in the middle of the Vicepit combo (ie a dungeon trash pack target dying mid-combo) it reverts to the filler combo, wasting the rest of the combo even though other enemies may still be within AoE range. By retaining the range check for Vicepit itself, the opportunity to waste the combo without enemies in range is still mitigated.

Akechi-kun reviewed (commented):
👍🏽💯

Akechi-kun commented:
> currently if you lose your target while in the middle of the Vicepit combo (ie a dungeon trash pack target dying mid-combo) it reverts to the filler combo, wasting the rest of the combo even though other enemies may still be within AoE range if you don't realize what has occurred and continue pressing the button.

I see, didn't know this was a thing. I suppose these option additions would help as a band-aid fix for that specifically for now, but I'll do some more digging on that scenario.
